src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -2665,6 +2664,91 @@ let updateTopLevelMetadata (arcFiles: ArcFiles) =
         }
     )
 
+let fillSelectedBuildingBlocksWithOntologyAnnotation (ontologyAnnotation: OntologyAnnotation) =
+    Excel.run(fun context ->
+        promise {
+            let! result = tryGetActiveAnnotationTable context
+
+            match result with
+            | Some excelTable ->
+
+                let! selectedBuildingBlock = getSelectedBuildingBlock excelTable context
+                let selectedRange = context.workbook.getSelectedRange().load(U2.Case2 (ResizeArray[|"rowCount"; "rowIndex"|]))
+
+                let tableRange = excelTable.getRange()
+                let _ = tableRange.load(U2.Case2 (ResizeArray[|"values"|]))
+
+                do! context.sync().``then``(fun _ -> ())
+
+                let tableValues =
+                    tableRange.values
+                    |> Array.ofSeq
+                    |> Array.map (fun row ->
+                        row
+                        |> Array.ofSeq
+                        |> Array.map (fun column ->
+                            column
+                            |> Option.map string
+                            |> Option.defaultValue ""
+                            |> (fun s -> s.TrimEnd())
+                        )
+                    )
+
+                let tableHeaders = tableValues.[0]
+                let firstRow = selectedRange.rowIndex
+                let rowCount = selectedRange.rowCount
+
+                if firstRow = 0 && rowCount = 1 then return [InteropLogging.Msg.create InteropLogging.Error "You cannot fill the headers of an annotation table with terms!"]
+                else
+                    let rowCount =
+                        if firstRow = 0. then rowCount - 1.
+                        else rowCount
+                    let firstRow =
+                        if firstRow = 0. then 1.
+                        else firstRow
+
+                    let columnIndices = selectedBuildingBlock |> Array.ofSeq |> Array.map (fun (index, _) -> index)
+                    let columnHeaders = ARCtrl.Spreadsheet.ArcTable.helperColumnStrings |> Array.ofSeq
+
+                    let firstIndex = Array.head columnIndices
+                    let lastIndex = Array.last columnIndices
+
+                    let isUnit = Array.contains columnHeaders.[2] tableHeaders.[firstIndex..lastIndex] //Unit
+                    for rowIndex in firstRow..(firstRow + rowCount-1.) do
+                        
+                        columnIndices
+                        |> Array.iter (fun columnIndex ->
+                            match tableHeaders.[columnIndex] with
+                            | header when header = columnHeaders.[2] -> //Unit
```

Comment:
This match case looks familiar, was this used somewhere else?

Reply:
Yes, a similiar matching was used for the term rectify